gym-unity/gym_unity/envs/__init__.py

```diff
@@ -386,6 +385,10 @@ def _sanitize_info(self, step_result: BatchedStepResult) -> BatchedStepResult:
         # only cares about the ordering.
         for index, agent_id in enumerate(step_result.agent_id):
             if not self._previous_step_result.contains_agent(agent_id):
+                if step_result.done[index]:
```

Comment:
Just making the mark_agent_done() change without this means that we'll register the agent and run out of "slots" later.